main.py

```diff
@@ -948,19 +954,49 @@ def notify_switch_enabled(self, dpid):
         event = KytosEvent(name=name, content={'dpid': dpid})
         self.controller.buffers.app.put(event)
 
+    def notify_switch_links_enable(self, switch):
+        """Send an event to notify the status of a link in a switch"""
+        for interface in switch.interfaces.values():
+            link = self._get_link_from_interface(interface)
+            if link:
+                if link.status == EntityStatus.UP:
+                    self.notify_link_status_change(link, reason="link up")
+
     def notify_switch_disabled(self, dpid):
         """Send an event to notify that a switch is disabled."""
         name = 'kytos/topology.switch.disabled'
         event = KytosEvent(name=name, content={'dpid': dpid})
         self.controller.buffers.app.put(event)
 
+    def notify_switch_links_disable(self, switch):
+        """Send an event to notify the status of a link in a switch"""
+        for interface in switch.interfaces.values():
+            link = self._get_link_from_interface(interface)
+            if link:
+                self.notify_link_status_change(link, reason="link down")
+
     def notify_topology_update(self):
         """Send an event to notify about updates on the topology."""
         name = 'kytos/topology.updated'
         event = KytosEvent(name=name, content={'topology':
                                                self._get_topology()})
         self.controller.buffers.app.put(event)
 
+    def notify_interface_link_enable(self, interface):
+        """Send an event to notify the status of a link from
+        an interface."""
+        link = self._get_link_from_interface(interface)
+        if link:
+            if link.status == EntityStatus.UP:
+                self.notify_link_status_change(link, reason="link up")
```

Comment:
In `notify_link_up_if_status` could the sleep be after the `if` condition? If not why is it necessary in this order? Does it need to sleep because `link.status` changes takes time?

Reply:
It's needed to ensure that other NApps won't change the status (liveness could change or other future protocol integration as well) and `kytos/topology.link_up` is only notified again if it hasn't been notified within a time interval (this is a requirement that we had to cover)